### DISCUSSION TITLE
Can OpenDeck run custom code?

### DISCUSSION THREAD

**Discussion by armandhammer** [shanteacontrols/OpenDeck#67] [Q&A]
Beyond the answer of "Yes it can do anything because it's Open Source", can OpenDeck run custom code?

Like if I wanted to intercept incoming notes and pitch everything up 1 octave, then send the changed notes to Midi Out, is it possible to write a script or something to do this?

**Comment by paradajz** [comment]
Yes it can do anything because it's Open Source.

With that out of the way, I'd suggest something like Bome MIDI translator or something similar if you need to do remapping on the fly.
